qemu/tests/virtio_mem_ignore_shared.py

```diff
@@ -0,0 +1,96 @@
+                               test.log.info)
+    test_mem = params.get_numeric("mem", target_type=float)
+    params["stress_args"] = '--cpu 4 --io 4 --vm 2 --vm-bytes %fM' % float(test_mem * 0.8)
+    try:
+        stress_test = utils_test.VMStress(vm, "stress", params)
+        error_context.context("Stress test start", test.log.info)
+        stress_test.load_stress_tool()
+    except utils_test.StressError as guest_info:
+        test.error(guest_info)
+    # Migration
+    error_context.base_context("Set migrate capabilities and do migration",
+                               test.log.info)
+    capabilities = ast.literal_eval(params.get("migrate_capabilities",
+                                               "{'x-ignore-shared': 'on'}"))
+    mig_timeout = params.get_numeric("mig_timeout", 1200, float)
+    mig_protocol = params.get("migration_protocol", "tcp")
+    clone = vm.migrate(mig_timeout,
+                       mig_protocol,
+                       env=env,
+                       migrate_capabilities=capabilities,
+                       not_wait_for_migration=True)
+    try:
+        vm.wait_for_migration(10)
+    except VMMigrateTimeoutError:
+        vm.wait_for_migration(mig_timeout)
+
+    clone.resume()
+    error_context.context("Check the total ram migrated", test.log.info)
+    total_mem_migrated = str(vm.monitor.info("migrate")['ram']['total'])
+    total_mem_migrated = float(normalize_data_size(total_mem_migrated, 'M'))
+    test.log.debug("Total memory migrated: %f" % total_mem_migrated)
+
+    if total_mem_migrated > test_mem * 0.01:
+        test.error("Error, more memory than expected has been migrated!")
+
+    error_context.base_context("Test virtio-mem device on destination VM",
+                               test.log.info)
+    virtio_mem_model = 'virtio-mem-pci'
+    if '-mmio:' in params.get("machine_type"):
+        virtio_mem_model = 'virtio-mem-device'
+    vmem_dev = clone.devices.get_by_params({'driver': virtio_mem_model})[0]
+    device_id = vmem_dev.get_qid()
+    requested_size_vmem = params.get("requested-size_test_vmem0")
+    for requested_size in requested_size_vmem.split():
+        req_size_normalized = int(float(normalize_data_size(requested_size,
+                                                            'B')))
+        clone.monitor.qom_set(device_id, "requested-size", req_size_normalized)
+        time.sleep(45)
+        virtio_mem_utils.check_memory_devices(device_id,
+                                              requested_size,
+                                              threshold,
+                                              clone,
+                                              test)
+        virtio_mem_utils.check_numa_plugged_mem(0,
+                                                requested_size,
+                                                threshold,
+                                                clone,
+                                                test)
```

Comment:
BTW, we should consider other tmp resource cleaning since the framework does not help us clean the environment well.

Reply:
Included the destroy and unregister for the clone VM

Reply:
> BTW, we should consider other tmp resource cleaning since the framework does not help us clean the environment well.

@YongxueHong you mean anything related with the stress tool? or just clean everything, in general